util/src/main/java/io/kubernetes/client/Copy.java

```diff
@@ -116,9 +119,14 @@ public void copyDirectoryFromPod(String namespace, String pod, String srcPath, P
   public void copyDirectoryFromPod(
       String namespace, String pod, String container, String srcPath, Path destination)
```

Comment:
> public void copyDirectoryFromPod(String namespace, String pod, String container, String srcPath, Path destination, boolean fallbackToNonTarCopy)

how about adding a new static method w/ argument `boolean fallbackToNonTarCopy` and make the old method reuse the new one? so that users can opt-in this change on their need?

Reply:
Ok, so do we want to make the new method public??

Reply:
i think so

Reply:
Ok- please take a look at my latest commit.

Reply:
sry for being unclear. i was proposing something like:

```java

public void copyDirectoryFromPod(String namespace, String pod, String container, String srcPath, Path destination) 
        throws ApiException, IOException, CopyNotSupportedException {
    if (!isTarPresentInContainer(namespace, pod, container)) {
      throw new CopyNotSupportedException("Tar is not present in the target container"); // If tar is not present then fallback to slow process of copying
    }
    copyDirectoryFromPod(namespace, pod, container, srcPath, destination, false)
}

public void copyDirectoryFromPod(String namespace, String pod, String container, String srcPath, Path destination, boolean enableTarCompressing) 
        throws ApiException, IOException {
    if (!enableTarCompressing) {
        copyDirectoryFromPod(namespace, pod, container, srcPath, destination, true);
        return;
    }
    // the old implemention
}
```

what do you think?

Reply:
SGTM. I will make the change.